### PR TITLE
docs(release): update changelog for the 2024-04 release

### PR DESCRIPTION
**Pull request by alevenberg**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13987)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,12 +17,56 @@ the APIs in these libraries are stable, and are ready for production use.
...
+
+### [Pub/Sub](/google/cloud/pubsub/README.md)
+
+- docs(pubsub): add kinesis sample ([#13593](https://github.com/googleapis/google-cloud-cpp/pull/13593))
+
+### [Spanner](/google/cloud/spanner/README.md)
+
+- feat(spanner): add ExcludeTransactionFromChangeStreamsOption ([#13898](https://github.com/googleapis/google-cloud-cpp/pull/13898))
+- feat(spanner): add spanner::Value support for TypeCode::FLOAT32 ([#13862](https://github.com/googleapis/google-cloud-cpp/pull/13862))
+- feat(generator): add polling policy sample ([#13797](https://github.com/googleapis/google-cloud-cpp/pull/13797))
```
Comment (devbww) [inline CHANGELOG.md:40]: Remove (at least from Spanner section?).

Comment (alevenberg) [reply CHANGELOG.md:40]: Oops missed that one. Thanks Brad.

Comment (devbww) [reply CHANGELOG.md]: It looks like you removed the #13759 entry too (not that that's a huge deal).

Comment (alevenberg) [reply CHANGELOG.md]: Fixed

**Review by devbww — APPROVED** [review]

**Review by dbolduc — APPROVED** [review]
LGTM, although I am not sure where we fall on publicizing the currently experimental `GCS+gRPC` feature

**Pull request merged**